### PR TITLE
fix(terminateandreplace): choose new seed before termination old one

### PR DESCRIPTION
**Pull request by aleksbykov**
The issue that if we have one seed node, we first terminate it and
then try to add new node, but it will failed, because no seeds left in cluster.
As a solution,

1.check if node is single seed
2.if yes, then choose random node and set it as seed
3.terminate target seed node and replace with new one.
4.leave cluster as is with seed node choosen on step 2

related to
https://github.com/scylladb/scylla-cluster-tests/pull/3651

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by aleksbykov** [comment]
fix only to rbno branch

File: sdcm/nemesis.py [hunk 1]
```diff
@@ -2289,10 +2289,13 @@ def disrupt_remove_node_then_add_node(self):
         # node_to_remove must be different than node
         verification_node = random.choice([n for n in self.cluster.nodes if n is not node_to_remove])
 
-        # node_to_remove must not be the only seed in cluster
+        # node_to_remove is single/last seed in cluster, berfore
```
Comment (fgelcer) [inline sdcm/nemesis.py:2292]: tinny typo in the word "before"

**Review by fgelcer — APPROVED** [review]
LGTM

i would back port (or create a PR with this commit + the previous commit as well to master, as it might be useful situation to our master too)

File: sdcm/nemesis.py (hunk 1 above)
Comment (aleksbykov) [reply sdcm/nemesis.py:2292]: fixed

**Comment by aleksbykov** [comment]
@roydahan do we need such functionality on master?

**Comment by fgelcer** [comment]
> @roydahan do we need such functionality on master?

AFAICT, once the RBNO will be enabled by default on scylla.git master, we will need it on sct.git master as well

**Comment by roydahan** [comment]
It's better  to send the PR to master and I'll just rebase RBNO branch.
Eventually we would want to merge rbno branch to master.

**Comment by aleksbykov** [comment]
Prepare new PR for master branch : https://github.com/scylladb/scylla-cluster-tests/pull/3686

**Review by bentsi — APPROVED** [review]

**Pull request merged**